### DISCUSSION TITLE
How can i retrieve via Api Streamer's Broadcasts start and end time?

### DISCUSSION THREAD

**Discussion by honoluluman2** [AzuraCast/AzuraCast#6460] [Q&A] [answered]
Hello, 

I am trying to get via Api the Streamer's Broadcasts start and end time [https://prnt.sc/rO6caZ21uLcY](https://prnt.sc/rO6caZ21uLcY)
[https://www.azuracast.com/api/](https://www.azuracast.com/api/)

Could some one point me out if this is possible? 

Thank you very much for your time.

**Comment by Sakretsos** [comment]
Hello @honoluluman2,
Hope you are well!

As i make a fast search on the azuracast API, i see that the information you are searching for is on the bellow GET api call
```
/station/{station_id}/streamer/{id}
```

The output will be the bellow:

```
{
  "id": 0,
  "streamer_username": "dj_test",
  "streamer_password": "",
  "display_name": "Test DJ",
  "comments": "This is a test DJ account.",
  "is_active": true,
  "enforce_schedule": false,
  "reactivate_at": 1609480800,
  "schedule_items": [
    "string"
  ]
}
```

I hope this is what you are searching!

Regards,
Ioannis

**Comment by honoluluman2** [comment]
Thank you @Sakretsos for your help.

Unfortunately this is not what i actually need.
I need to found some API endpoint that will give me the Streamer's last (most recent) Broadcast start and end time.
From /station/{station_id}/streamer/{id} there is no such information.

From azura admin, on streamer's account i can see the logs [https://prnt.sc/rO6caZ21uLcY](https://prnt.sc/rO6caZ21uLcY).
But is there some endpoint that could give me the last (most recent) log? (maybe as a timestamp)

**Comment by Moonbase59** [comment, marked as answer]
Use the `/station/{station_id}/streamer/{id}` API endpoint to get the `links.broadcasts` entry. If it exists, it looks like

```
https://example.com/api/station/{station_id}/streamer/{id}/broadcasts
```

as in this example:

```json
{
  "streamer_username": "moonbase",
  "streamer_password": "",
  "display_name": "Moonbase",
  "comments": null,
  "is_active": true,
  "enforce_schedule": false,
  "reactivate_at": null,
  "art_updated_at": 1678618644,
  "schedule_items": [],
  "id": 1,
  "links": {
    "self": "https://example.com/api/station/1/streamer/1",
    "broadcasts": "https://example.com/api/station/1/streamer/1/broadcasts",
    "art": "https://example.com/api/station/1/streamer/1/art"
  },
  "has_custom_art": true,
  "art": "https://example.com/api/station/1/streamer/1%7C1678618644/art"
}
```

Using a GET returns a JSON list of that streamer’s broadcasts, like so:

```json
[
  {
    "timestampStart": 1690096103,
    "timestampEnd": 1690096419,
    "id": 92,
    "recording": [],
    "links": {
      "delete": "https://example.com/api/station/1/streamer/1/broadcast/92"
    }
  },
  {
    "timestampStart": 1689764573,
    "timestampEnd": 1689764890,
    "id": 91,
    "recording": [],
    "links": {
      "delete": "https://example.com/api/station/1/streamer/1/broadcast/91"
    }
  }
]
```

**Note:** You will need an _API authorization key_ with appropriate rights to get at this information.

**Note 2:** Timestamps are in UTC, so you might need to convert them to your station’s local time. (See API `/admin/station/{id}` → `timezone`.)

Try it out using your station’s _Swagger UI_ at `https://example.com/static/api/index.html#/Stations:%20Streamers/DJs/getStreamer`.

**Comment by honoluluman2** [comment]
thank you very much @Moonbase59 this is what i needed.
